File: mbw/src/main/java/com/mycelium/wallet/content/HandleConfigFactory.java
package com.mycelium.wallet.content;


import com.mycelium.wallet.content.actions.AddressAction;
import com.mycelium.wallet.content.actions.BitIdAction;
import com.mycelium.wallet.content.actions.HdNodeAction;
import com.mycelium.wallet.content.actions.MasterSeedAction;
import com.mycelium.wallet.content.actions.PopAction;
import com.mycelium.wallet.content.actions.PrivateKeyAction;
import com.mycelium.wallet.content.actions.SssShareAction;
import com.mycelium.wallet.content.actions.UriAction;
import com.mycelium.wallet.content.actions.WebsiteAction;
import com.mycelium.wallet.content.actions.WordListAction;

public class HandleConfigFactory {
    public static StringHandleConfig returnKeyOrAddressOrUriOrKeynode() {
        StringHandleConfig request = new StringHandleConfig();
        request.privateKeyAction = new PrivateKeyAction();
        request.addressAction = new AddressAction();
        request.bitcoinUriAction = new UriAction();
        request.hdNodeAction = new HdNodeAction();
        request.popAction = new PopAction();
        return request;
    }

    public static StringHandleConfig returnKeyOrAddressOrHdNode() {
        StringHandleConfig request = new StringHandleConfig();
        request.privateKeyAction = new PrivateKeyAction();
        request.hdNodeAction = new HdNodeAction();
        request.addressAction = new AddressAction();
        request.bitcoinUriAction = new UriAction(true);
        request.sssShareAction = new SssShareAction();
        return request;
    }

    public static StringHandleConfig spendFromColdStorage() {
        StringHandleConfig request = new StringHandleConfig();
        request.privateKeyAction = new PrivateKeyAction();
        request.addressAction = new AddressAction();
        request.bitcoinUriAction = new UriAction(true);
        request.hdNodeAction = new HdNodeAction();
        request.sssShareAction = new SssShareAction();
        request.wordListAction = new WordListAction();
        return request;
    }

    public static StringHandleConfig getAddressBookScanRequest() {
        StringHandleConfig request = new StringHandleConfig();
        request.privateKeyAction = new PrivateKeyAction();
        request.addressAction = new AddressAction();
        request.bitcoinUriAction = new UriAction(true);
        return request;
    }

    public static StringHandleConfig genericScanRequest() {
        StringHandleConfig request = new StringHandleConfig();
        request.addressAction = new AddressAction();
        request.bitcoinUriAction = new UriAction();
        request.bitIdAction = new BitIdAction();
        request.privateKeyAction = new PrivateKeyAction();
        request.websiteAction = new WebsiteAction();
        request.sssShareAction = new SssShareAction();
        request.wordListAction = new WordListAction();
        request.hdNodeAction = new HdNodeAction();
        request.popAction = new PopAction();

        //at the moment, we just support wordlist backups
        //request.masterSeedAction = MasterSeedAction.IMPORT;
        return request;
    }

    public static StringHandleConfig getShare() {
        StringHandleConfig request = new StringHandleConfig();
        request.sssShareAction = new SssShareAction();
        return request;
    }

    public static StringHandleConfig verifySeedOrKey() {
        StringHandleConfig request = new StringHandleConfig();
        request.masterSeedAction = new MasterSeedAction();
        request.privateKeyAction = new PrivateKeyAction();
        return request;
    }
}

File: mbw/src/main/java/com/mycelium/wallet/content/HandleConfigFactory.kt
package com.mycelium.wallet.content


import com.mycelium.wallet.content.actions.AddressAction
import com.mycelium.wallet.content.actions.BitIdAction
import com.mycelium.wallet.content.actions.HdNodeAction
import com.mycelium.wallet.content.actions.MasterSeedAction
import com.mycelium.wallet.content.actions.PopAction
import com.mycelium.wallet.content.actions.PrivateKeyAction
import com.mycelium.wallet.content.actions.SssShareAction
import com.mycelium.wallet.content.actions.UriAction
import com.mycelium.wallet.content.actions.WebsiteAction
import com.mycelium.wallet.content.actions.WordListAction

object HandleConfigFactory {
    @JvmStatic
    val addressBookScanRequest = StringHandleConfig().apply {
        privateKeyAction = PrivateKeyAction()
        addressAction = AddressAction()
        bitcoinUriAction = UriAction(true)
    }

    @JvmStatic
    val share = StringHandleConfig().apply {
        sssShareAction = SssShareAction()
    }

    @JvmStatic
    fun returnKeyOrAddressOrUriOrKeynode() = StringHandleConfig().apply {
        privateKeyAction = PrivateKeyAction()
        addressAction = AddressAction()
        bitcoinUriAction = UriAction()
        hdNodeAction = HdNodeAction()
        popAction = PopAction()
    }

    @JvmStatic
    fun returnKeyOrAddressOrHdNode() = StringHandleConfig().apply {
        privateKeyAction = PrivateKeyAction()
        hdNodeAction = HdNodeAction()
        addressAction = AddressAction()
        bitcoinUriAction = UriAction(true)
        sssShareAction = SssShareAction()
    }

    @JvmStatic
    fun spendFromColdStorage() = StringHandleConfig().apply {
        privateKeyAction = PrivateKeyAction()
        addressAction = AddressAction()
        bitcoinUriAction = UriAction(true)
        hdNodeAction = HdNodeAction()
        sssShareAction = SssShareAction()
        wordListAction = WordListAction()
    }

    @JvmStatic
    fun genericScanRequest() = StringHandleConfig().apply {
        addressAction = AddressAction()
        bitcoinUriAction = UriAction()
        bitIdAction = BitIdAction()
        privateKeyAction = PrivateKeyAction()
        websiteAction = WebsiteAction()
        sssShareAction = SssShareAction()
        wordListAction = WordListAction()
        hdNodeAction = HdNodeAction()
        popAction = PopAction()
        //at the moment, we just support wordlist backups
        //masterSeedAction = MasterSeedAction.IMPORT;
    }

    @JvmStatic
    fun verifySeedOrKey() = StringHandleConfig().apply {
        masterSeedAction = MasterSeedAction()
        privateKeyAction = PrivateKeyAction()
    }
}